arxiv/taxonomy/taxonomy_test.py

```diff
@@ -168,4 +168,55 @@ def test_bad_objects(self):
         self.assertEqual(cat.full_name, "Invalid Category: physics", "full name should have data on original bad item")
         self.assertEqual(cat.full_name, cat.get_canonical().full_name, "name data should be retained after canonical calls")
         self.assertEqual(arch.full_name, "Invalid Archive: hamsters", "full name should have data on original bad item")
-        self.assertEqual(arch.full_name, arch.get_canonical().full_name, "name data should be retained after canonical calls")
\ No newline at end of file
+        self.assertEqual(arch.full_name, arch.get_canonical().full_name, "name data should be retained after canonical calls")
+
+    def test_all_cats_from_string(self):
+        #basic
```

Comment:
Would be good to add an empty string, string with only spaces and a string that is nonsense.